lib/src/rules/prefer_final_fields.dart

```diff
@@ -112,16 +99,39 @@ class PreferFinalFields extends LintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    var visitor = _Visitor(this);
+    var visitor = _Visitor(this, context);
     registry.addCompilationUnit(this, visitor);
-    registry.addFieldDeclaration(this, visitor);
+  }
+}
+
+class _DeclarationsCollector extends RecursiveAstVisitor<void> {
+  final fields = <FieldElement, VariableDeclaration>{};
+
+  @override
+  void visitFieldDeclaration(FieldDeclaration node) {
+    if (node.parent is EnumDeclaration) return;
+    if (node.fields.isFinal || node.fields.isConst) {
+      return;
+    }
+
+    for (var variable in node.fields.variables) {
+      var element = variable.declaredElement;
+      if (element is FieldElement &&
+          element.isPrivate &&
+          !element.overridesField) {
+        fields[element] = variable;
+      }
+    }
   }
 }
 
 class _MutatedFieldsCollector extends RecursiveAstVisitor<void> {
```

Comment:
nit: This isn't collecting mutated fields anymore (which is a good thing), it's finding fields that have been mutated and updating the list. Perhaps the name could be updated to reflect the change. I'm sure you can come up with something better, but I'm thinking of something like `_FieldMutationFinder`.

Reply:
That sounds great, and doesn't change the alphabetical ordering 😌

Reply:
I'll try to keep that goal in mind when I'm renaming other classes. 🙂